packages/@react-spectrum/combobox/src/ComboBox.tsx

```diff
@@ -137,6 +137,17 @@ const ComboBoxBase = React.forwardRef(function ComboBoxBase<T extends object>(pr
 
   useLayoutEffect(onResize, [scale, onResize]);
 
+  // Update position once the ListBox has rendered. This ensures that
+  // it flips properly when it doesn't fit in the available space.
+  // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
+  useLayoutEffect(() => {
+    if (state.isOpen) {
+      requestAnimationFrame(() => {
+        updatePosition();
+      });
+    }
+  }, [state.isOpen, updatePosition]);
```

Comment:
Taken straight from Picker. I thought I remembered this bit of code causing problems in the past for ComboBox but I dug through the repo/pulls/issues and couldn't find anything. Seems to work fine in testing